File: apple/Bennett/Models/TriggerEvent.swift
import Foundation

enum TriggerEvent: String, Codable, Sendable {
    case appOpen = "app_open"
    case doubleLockComplete = "double_lock_complete"
    case streakMilestone = "streak_milestone"
    case streakSlip = "streak_slip"
    case shieldIgnored3x = "shield_ignored_3x"
    case doerTabEnter = "doer_tab_enter"
    case moduleComplete = "module_complete"
    case energyCheck = "energy_check"
    case betaFeedbackPrompt = "beta_feedback_prompt"
    case redemptionQuestTrigger = "redemption_quest_trigger"
    case refundEarned = "refund_earned"
    case doerTabLocked = "doer_tab_locked"
    case doerTabUnlocked = "doer_tab_unlocked"
    case onboarding
}

File: shared/types/TriggerEvent.ts
export type TriggerEvent =
  | 'app_open'
  | 'double_lock_complete'
  | 'streak_milestone'
  | 'streak_slip'
  | 'shield_ignored_3x'
  | 'doer_tab_enter'
  | 'module_complete'
  | 'energy_check'
  | 'beta_feedback_prompt'
  | 'redemption_quest_trigger'
  | 'refund_earned'
  | 'doer_tab_locked'
  | 'doer_tab_unlocked'
  | 'onboarding'
  | 'otf_prompt'
  | 'psm_success'
  | 'psm_failure';

export const TRIGGER_EVENTS: readonly TriggerEvent[] = [
  'app_open',
  'double_lock_complete',
  'streak_milestone',
  'streak_slip',
  'shield_ignored_3x',
  'doer_tab_enter',
  'module_complete',
  'energy_check',
  'beta_feedback_prompt',
  'redemption_quest_trigger',
  'refund_earned',
  'doer_tab_locked',
  'doer_tab_unlocked',
  'onboarding',
  'otf_prompt',
  'psm_success',
  'psm_failure',
] as const;
